setup.cfg

```diff
@@ -8,3 +8,7 @@ test = trial
 
 [trial]
 test_suite = tests
+
+[flake8]
+max-line-length = 88
+extend-ignore = E203,E501,F821 # TODO: Fix E501 and F821
```

Comment:
E501 is line length; we can deal with that separately.

F821 is undefined name... this occurs on [two lines in sydent/http/federation_tls_options.py](https://github.com/matrix-org/sydent/blob/f45a55c300dcd2f77c43408a649184c088ffed9d/sydent/http/federation_tls_options.py#L37-L38), and frankly I just need more time to understand context here, since that code cannot work as it is, but has been in place for a few years.

Reply:
Re: F821- so if a piece of code is broken but it's been there awhile the prudent thing to do is investigate it thoroughly rather than just fix/remove?

Reply:
Yep! See [Chesterton's fence](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence) -- I don't know _why_ that code is like that, and until I do, it's hard to know _how_ it should be fixed. What I _do_ know is that Sydent has been running with that code present for literally two years, so it can't be too problematic. So why?

Maybe its enclosing function is dead code? (Nope; I just checked and several paths eventually lead to it).

It's in an exception handler, so maybe the code it's wrapping never actually raises an exception, so control never passes into the `except` block? (Unlikely...)

Or maybe when it execution does flow into the `except` block it just... blows up with a NameError instead of properly handling things. (Likely!) Looking at the [upstream Twisted code](https://github.com/twisted/twisted/blob/7c84d1d92ca895059ed6c4251cac62e5badd36ef/src/twisted/internet/_sslverify.py#L1041) that this borrows from, it looks like its purpose is to just do a better job of logging an error, and then drop a connection... so quite possibly benign if it fails.